xsplice: Use --build-id=sha1 consistently

Pass --build-id=sha1 to every invocation of ld to ensure that symbol
offsets are correct.

diff --git a/xen/arch/x86/Makefile b/xen/arch/x86/Makefile
index 18ff0c6..6f18a10 100644
--- a/xen/arch/x86/Makefile
+++ b/xen/arch/x86/Makefile
@@ -119,10 +119,10 @@ $(BASEDIR)/common/symbols-dummy.o:
 
 $(TARGET)-syms: prelink.o xen.lds $(BASEDIR)/common/symbols-dummy.o efi/relocs-dummy.o
 	$(LD) $(LDFLAGS) -T xen.lds -N prelink.o efi/relocs-dummy.o \
-	    $(BASEDIR)/common/symbols-dummy.o -o $(@D)/.$(@F).0
+	    $(BASEDIR)/common/symbols-dummy.o $(build_id_linker) -o $(@D)/.$(@F).0
 	$(NM) -pa --format=sysv $(@D)/.$(@F).0 | $(BASEDIR)/tools/symbols --all-symbols --sysv --sort >$(@D)/.$(@F).0.S
 	$(MAKE) -f $(BASEDIR)/Rules.mk $(@D)/.$(@F).0.o
-	$(LD) $(LDFLAGS) -T xen.lds -N prelink.o efi/relocs-dummy.o \
+	$(LD) $(LDFLAGS) -T xen.lds -N prelink.o $(build_id_linker) efi/relocs-dummy.o \
 	    $(@D)/.$(@F).0.o -o $(@D)/.$(@F).1
 	$(NM) -pa --format=sysv $(@D)/.$(@F).1 | $(BASEDIR)/tools/symbols --all-symbols --sysv --sort --warn-dup >$(@D)/.$(@F).1.S
 	$(MAKE) -f $(BASEDIR)/Rules.mk $(@D)/.$(@F).1.o
